battery.sh

```diff
@@ -487,8 +500,15 @@ if [[ "$action" == "maintain" ]]; then
 		exit 0
 	fi
 
+	if [[ "$setting" =~ ^[0-9]{1,3}-[0-9]{1,3}$ ]];then
```

Comment:
Not sure how anyone would ever enter 100 or higher in either lower nor upper range, but here is an improvement suggestion:

^[0-9]{1,3}-(100|[0-9]{1,2})$

You could also do 1+[0-9]{1,2} and check against 101-199 but I thought it not as elegant :)

Reply:
You inspired me. 
All inputs that match the range mode should be matched, and user should be prompted when the number does not meet the requirements. That is better.